include/seqan3/argument_parser/validators.hpp

```diff
@@ -351,7 +351,8 @@ class file_validator_base
      */
     template <std::ranges::forward_range range_type>
     //!\cond
-        requires std::convertible_to<std::ranges::range_value_t<range_type>, std::filesystem::path const &>
+        requires (std::convertible_to<std::ranges::range_value_t<range_type>, std::filesystem::path const &>
+                 && !std::is_base_of_v<std::filesystem::path, range_type>)
```

Comment:
Without having too much thought about this, but this seems to be a code smell and might need a different solution.

My initial thought is that the `std::filesystem::path` is one of those rare cases that have infinite range depth as each value is itself a `std::filesystem::path`.

The original intend is to allow a range of "std::filesystem::path"-like objects and convert them.

This one is tough...

Reply:
Yes, I think there is too much magic going on :/
The solution of using `!std::convertible_to<range_type, std::filesystem::path const &>` is much better than mine.